server/src/internalClusterTest/java/org/opensearch/search/geo/GeoFilterIT.java

```diff
@@ -115,20 +115,19 @@ public static void createNodes() throws Exception {
     }
 
     private static byte[] unZipData(String path) throws IOException {
-        try (InputStream is = Streams.class.getResourceAsStream(path)) {
-            if (is == null) {
-                throw new FileNotFoundException("Resource [" + path + "] not found in classpath");
-            }
+        InputStream is = Streams.class.getResourceAsStream(path);
+        if (is == null) {
+            throw new FileNotFoundException("Resource [" + path + "] not found in classpath");
+        }
 
-            try (
-                ByteArrayOutputStream out = new ByteArrayOutputStream();
-                GZIPInputStream in = new GZIPInputStream(is)
-            ) {
-                Streams.copy(in, out);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        GZIPInputStream in = new GZIPInputStream(is);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *in*. The resource is referenced at lines 119, 124, 125. The resource is closed at line 127. There are other execution paths that do not contain closure statements, for example, when Streams.copy throws an exception. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).